src/traces/volume/convert.js

```diff
@@ -161,8 +182,6 @@ function convert(gl, scene, trace) {
         trace.imax === undefined ? trace.cmax : trace.imax
     ];
 
-    volumeOpts.opacity = trace.opacity === undefined ? 1 : trace.opacity;
```

Comment:
You can still set this, if it'll get used downstream - which it looks like it will based on your amended description for `opacityscale`, thanks for fleshing that out!

My point was just that you don't need to handle the `undefined` case here, since the global `trace.opacity` attribute has a default of 1 you know at this point that you can only have a number between 0 and 1.

Reply:
Ohh, right, thanks for catching this.